server/routes/api/form.js

```diff
@@ -0,0 +1,195 @@
+var router = require('express').Router(),
+    fs = require('fs'),
+    path = require('path'),
+    config = require('../../../config/default.js'),
+    authMiddleware = require('../../middleware/auth.js');
+
+function handler(req, res) {
```

Comment:
Also can you add some comments just to clarify whats happening for future readers, this code is pretty dense